ksql-functional-tests/src/main/java/io/confluent/ksql/test/tools/KsqlTestingTool.java

```diff
@@ -0,0 +1,119 @@
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.test.tools;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.confluent.ksql.function.TestFunctionRegistry;
+import io.confluent.ksql.json.JsonMapper;
+import io.confluent.ksql.test.model.QttTestFile;
+import io.confluent.ksql.test.model.TestCaseNode;
+import io.confluent.ksql.test.tools.command.TestOptions;
+import io.confluent.ksql.util.Pair;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public final class KsqlTestingTool {
+
+  private KsqlTestingTool() {
+
+  }
+
+  private static final ObjectMapper OBJECT_MAPPER = JsonMapper.INSTANCE.mapper;
+
+  private static int totalNumberOfTests;
+
+  public static void main(final String[] args) throws IOException {
+    loadAndRunTests(args);
+  }
+
+  static void loadAndRunTests(final String[] args) {
+    totalNumberOfTests = 0;
+    final List<String> passedTests = new ArrayList<>();
+    final List<Pair<String, String>> failedTests = new ArrayList<>();
+    try {
+
+      final TestOptions testOptions = TestOptions.parse(args);
+
+      if (testOptions == null) {
+        System.err.println("Invalid parameter.");
+        System.err.println("Usage: ksql-testing-tool /path/to/the/test/file.json");
+        return;
+      }
+
+      final QttTestFile qttTestFile = OBJECT_MAPPER.readValue(
+          new File(testOptions.getTestFile()), QttTestFile.class);
+      for (final TestCaseNode testCaseNode: qttTestFile.tests) {
+        final List<TestCase> testCases = testCaseNode.buildTests(
+            new File(testOptions.getTestFile()).toPath(),
+            TestFunctionRegistry.INSTANCE.get());
+        for (final TestCase testCase: testCases) {
+          executeTestCase(
+              testCase,
+              new TestExecutor(),
+              passedTests,
+              failedTests);
+        }
+      }
+      printResults(totalNumberOfTests, passedTests, failedTests);
+    } catch (final Exception e) {
+      System.err.println("Failed to start KSQL testing tool: " + e.getMessage());
+    }
+  }
+
+  static void executeTestCase(
+      final TestCase testCase,
+      final TestExecutor testExecutor,
+      final List<String> passedTests,
+      final List<Pair<String, String>> failedTests
+  ) {
+    try {
+
+      System.out.println(" >>> Running test: " + testCase.getName());
+      testExecutor.buildAndExecuteQuery(testCase);
+      System.out.println("\t >>> Test " + testCase.getName() + " passed!");
+      passedTests.add(testCase.getName());
+      totalNumberOfTests ++;
```

Comment:
This needs to happen either at the beginning of the try-catch block, or in the `finally` block. Currently only successful tests are being added to the count.

Reply:
Oops, you are right. Moved it to finally block.